### PR TITLE
<__msvc_all_public_headers.hpp> Silence experimental erase

### PR DESCRIPTION
**Pull request by SuperWig**
# Description

Seems like this should be here.

# Checklist

Be sure you've read README.md and understand the scope of this repo.

If you're unsure about a box, leave it unchecked. A maintainer will help you.

- [x] Identifiers in product code changes are properly `_Ugly` as per
  https://eel.is/c++draft/lex.name#3.1 or there are no product code changes.
- [ ] The STL builds successfully and all tests have passed (must be manually
  verified by an STL maintainer before automated testing is enabled on GitHub,
  leave this unchecked for initial submission).
- [x] These changes introduce no known ABI breaks (adding members, renaming
  members, adding virtual functions, changing whether a type is an aggregate
  or trivially copyable, etc.).
- [x] These changes were written from scratch using only this repository,
  the C++ Working Draft (including any cited standards), other WG21 papers
  (excluding reference implementations outside of proposed standard wording),
  and LWG issues as reference material. If they were derived from a project
  that's already listed in NOTICE.txt, that's fine, but please mention it.
  If they were derived from any other project (including Boost and libc++,
  which are not yet listed in NOTICE.txt), you *must* mention it here,
  so we can determine whether the license is compatible and what else needs
  to be done.

### CODE REVIEW

File: stl/inc/__msvc_all_public_headers.hpp [hunk 1]
```diff
@@ -34,6 +34,7 @@
 
 #define _SILENCE_CXX17_C_HEADER_DEPRECATION_WARNING
 #define _SILENCE_CXX17_STRSTREAM_DEPRECATION_WARNING
+#define _SILENCE_EXPERIMENTAL_ERASE_DEPRECATION_WARNING
```
Comment (CaseyCarter) [inline stl/inc/__msvc_all_public_headers.hpp]: The purpose of these macro definitions is to suppress errors/warnings as a consequence of including headers. (We think of these as "aggressive deprecations".) There is no header for which `_SILENCE_EXPERIMENTAL_ERASE_DEPRECATION_WARNING` suppresses such an error/warning; its definition only makes it possible to *use* the `std::experimental::erase_if` without a deprecation warning.

As such, I don't think we should make this change.

Comment (SuperWig) [reply stl/inc/__msvc_all_public_headers.hpp]: Ahhh, thanks for the explanation.

Comment (StephanTLavavej) [reply stl/inc/__msvc_all_public_headers.hpp]: Agreed. By not silencing the `experimental::erase_if` deprecation here, our test coverage verifies that nothing in the STL is using `experimental::erase_if`. (The tests specifically for `experimental::erase_if` then silence the warning.) Of course, there's no way you could have known that without seeing our tests (which will be available soon!). :-)

**Comment by StephanTLavavej** [comment]
As Casey explained, this was intentional, so I'm going to go ahead and close this pull request. Thanks for looking into this, though - inconsistencies are sometimes oversights!

If you feel that this deserves a comment explaining that only the header-wide deprecations are being silenced, I would be happy to merge such a PR.

**Pull request closed without merging**